creusot/src/analysis.rs

```diff
@@ -901,14 +901,17 @@ fn translate_vars<'tcx>(
 ) -> (fmir::LocalDecls<'tcx>, HashMap<Local, (Symbol, Ident)>) {
     let mut vars = fmir::LocalDecls::with_capacity(body.local_decls.len());
     let mut locals = HashMap::new();
+    let external_body = !body.source.def_id().is_local();
 
     use mir::VarDebugInfoContents::Place;
 
     for (loc, d) in body.local_decls.iter_enumerated() {
         if erased_locals.contains(loc) {
             continue;
         }
-        let name = if !d.is_user_variable() {
+        // `is_user_variable` panics if the body comes from another crate
+        let temp = external_body || !d.is_user_variable();
```

Comment:
`constant_propagation.rs` apparently makes use of this information but I'm not sure what the impact is of marking all variables as "temporary" in const definitions from other crates.

For now I have yet to come across a `const` where this matters.